packages/react-router/lib/dom/ssr/fog-of-war.ts

```diff
@@ -116,14 +116,23 @@ export function useFogOFWarDiscovery(
       if (!path) {
         return;
       }
-      let url = new URL(path, window.location.origin);
-      if (!discoveredPaths.has(url.pathname)) {
-        nextPaths.add(url.pathname);
+      // optimization: use the already-parsed pathname from links
+      let pathname =
+        el.tagName === 'A'
+          ? (el as HTMLAnchorElement).pathname
```

Comment:
It almost certainly doesn't matter, but `<area>` also has these helpers

Reply:
this only runs on `<a>` and `<form>` (I assume because remix doesn't support other types of links?)